files/image_config/sysmonitor/sysmonitor.py

```diff
@@ -0,0 +1,691 @@
+    
+    dnsrvs_name=str1.join(dnsrvs_name_list)
+    sysready_lock.unlock()
+    
+    return {"status":msg, "allsrvs":allsrvs, "dnsrvs_name":dnsrvs_name}
+
+
+def sysready_listen():
+    SERVER_ADDRESS = '/var/run/sysready.socket'
+
+    if os.path.exists(SERVER_ADDRESS):
+        os.remove(SERVER_ADDRESS)
+
+    sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    sock.bind(SERVER_ADDRESS)
+    os.chmod(SERVER_ADDRESS, stat.S_IRWXU |stat.S_IRGRP | stat.S_IWGRP | stat.S_IROTH | stat.S_IWOTH)
+    sock.listen(1)
+    fail_res={"status":False, "msg":None}
+    while True:
+        connection, client_address = sock.accept()
+        try:
+            request = connection.recv(10240)
+            #logger.log_info("sysready [ REQ ] {}".format(request))
+            if request is None:
+                continue
+
+            req=Dict2Obj(json.loads(request.decode('utf-8')))
+
+            response = globals()[req.command](req)
+            res=json.dumps(response)
+            #logger.log_info("sysready [ RES ] {}".format(res))
+            connection.sendall(res.encode('utf-8'))
+        except Exception as e:
+            logger.log_error("sysready {}".format(str(e)))
+            fail_res['msg']=str(e)
+            connection.sendall(json.dumps(fail_res).encode('utf-8'))
+
+        connection.close()
+
+    #sock.close() #lgtm [py/unreachable-statement]
+
+
+def db_connect():
+    try:
+        st_db = SonicV2Connector()
+        st_db.connect(st_db.STATE_DB,True)
+        ap_db = SonicV2Connector()
+        ap_db.connect(ap_db.APPL_DB,True)
+        config_db = ConfigDBConnector()
+        config_db.connect()
+    except Exception as e:
+        logger.log_error("Error: Connection to the DB failed {}".format(str(e)))
+        sys.exit(1)
+
+    return st_db,ap_db,config_db
+
+
+def system_service():
+
+    global QUEUE
+    QUEUE = mp.Queue()
+
+    st_db,ap_db,config_db = db_connect()
+
+    thread_service_event = threading.Thread(target=subscribe_service_event_thread, name='service', args=(QUEUE,))
+    thread_service_event.start()
+
+    thread_sysready = threading.Thread(target=sysready_listen, name='sysready', args=())
+    thread_sysready.start()
+
+    thread_statedb = threading.Thread(target=subscribe_statedb_event_thread, name='statedb', args=(QUEUE,))
+    thread_statedb.start()
+
+    event = 'SERVICE_EVENT'
+    sysready_lock.lock()
+    #This is run only once when sysmonitor bootsup
+    check_system_status(event, st_db, ap_db, config_db)
+    sysready_lock.unlock()
+
+    # Queue to receive the STATEDB and Systemd state change event
+    while True:
+        event = QUEUE.get()
+        #logger.log_info( "System event [ "+event+" ] is received")
+        try:
+            sysready_lock.lock()
+            check_unit_status(event, st_db, ap_db, config_db)
+            sysready_lock.unlock()
+        except Exception as e:
+            logger.log_error( str(e))
+            time.sleep(2)
```

Comment:
Do we need to periodically check the status every 2 secs? what's the CPU usage for this service?

Reply:
sleep(2) is invoked only in case of exception. 
There is no periodic check. 
As the design is on event based framework, the logic is to check the status of a service unit that was received in the queue. It will not hog the cpu.